controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -444,11 +678,25 @@ func (w *Workload) updateStaticPodCondition(ctx context.Context, machine *cluste
 		if len(containerWaitingMessages) > 0 {
 			if terminatedWithError {
 				conditions.MarkFalse(machine, staticPodCondition, controlplanev1.PodFailedReason, clusterv1.ConditionSeverityError, strings.Join(containerWaitingMessages, ", "))
+
+				v1beta2conditions.Set(machine, metav1.Condition{
+					Type:    staticPodV1beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  controlplanev1.KubeadmControlPlaneMachinePodFailedV1Beta2Reason,
+					Message: strings.Join(containerWaitingMessages, ", "),
+				})
 				return
 			}
 			// Note: Some error cases cannot be caught when container state == "Waiting",
 			// e.g., "waiting.reason: ErrImagePull" is an error, but since LastTerminationState does not exist, this cannot be differentiated from "PodProvisioningReason"
 			conditions.MarkFalse(machine, staticPodCondition, controlplanev1.PodProvisioningReason, clusterv1.ConditionSeverityInfo, strings.Join(containerWaitingMessages, ", "))
+
+			v1beta2conditions.Set(machine, metav1.Condition{
+				Type:    staticPodV1beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  controlplanev1.KubeadmControlPlaneMachinePodProvisioningV1Beta2Reason,
+				Message: "",
```

Comment:
Is it intentional that the message is empty here while we keep surfacing the same in l.686?

Reply:
Good catch!